SilKit/source/experimental/netsim/SimulatedNetworkInternal.cpp

```diff
@@ -57,21 +58,33 @@ void SimulatedNetworkInternal::CreateAndSetEventProducer()
     }
 }
 
+auto SimulatedNetworkInternal::ExtractControllerTypeName(const SilKit::Core::ServiceDescriptor& serviceDescriptor)
```

Comment:
since this is only used in `SimulatedNetworkRouter::AddSimulatedController`, i wonder if we could just move it down inside the call of AddSimulatedController.

Reply:
I wanted to isolate the "out parameter" and possible execption that happens there just for readability.